### PR TITLE
feat: implement tms schematics

### LINKED ISSUE #11501: TMS schematics
This ticket is about adding TMS schematics.

### PR DESCRIPTION
**Pull request by znikola**
closes #11501

### CODE REVIEW

File: feature-libs/tracking/schematics/add-tracking/schema.json [hunk 1]
```diff
@@ -13,8 +13,26 @@
...
+      "description": "Lazy load the tracking features.",
       "default": true
+    },
+    "features": {
+      "type": "array",
+      "uniqueItems": true,
+      "items": {
+        "enum": [
+          "Personalization",
+          "Google Tag Manager",
```
Comment (marlass) [inline feature-libs/tracking/schematics/add-tracking/schema.json]: What do you think about prefixing them with tag management?

eg. `Tag management / Google Tag Manager`, so we see that those 2 are part of "tag managers" family?

Now it looks like any of those features are top level feature, but in reality it is personalization / tag management

File: feature-libs/tracking/schematics/add-tracking/index.ts [hunk 2]
```diff
@@ -60,13 +196,13 @@ function addPersonalizationFeature(
...
 
-function addPersonalizationPackageJsonDependencies(packageJson: any): Rule {
+function addTrackingPackageJsonDependencies(packageJson: any): Rule {
   const spartacusVersion = `^${getSpartacusSchematicsVersion()}`;
   const dependencies: NodeDependency[] = [
     {
       type: NodeDependencyType.Default,
       version: spartacusVersion,
-      name: SPARTACUS_PERSONALIZATION,
+      name: SPARTACUS_TRACKING,
```
Comment (marlass) [inline feature-libs/tracking/schematics/add-tracking/index.ts]: Do we need to specify dependency here?

Option with
```
"ng-add": {
    "save": "dependencies"
  },
```

imho is cleaner

File: feature-libs/tracking/schematics/add-tracking/schema.json (hunk 1 above)
Comment (znikola) [reply feature-libs/tracking/schematics/add-tracking/schema.json]: Done

File: feature-libs/tracking/schematics/add-tracking/index.ts (hunk 2 above)
Comment (znikola) [reply feature-libs/tracking/schematics/add-tracking/index.ts]: Agreed, we don't have to add this dep manually.
We don't even need 
```json
"ng-add": {
    "save": "dependencies"
  },
```
in the tracking's `package.json`, it's being added by default to `dependencies`.

**Review by marlass — APPROVED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**